plugin/evm/message/handshake/upgrade_config_message.go

```diff
@@ -36,45 +24,32 @@ func (u *UpgradeConfigMessage) ID() common.Hash {
 	return u.hash
 }
 
-// Attempts to parse a networkUpgradeConfigMessage from a []byte
-//
-// This function attempts to parse a stream of bytes as a
-// networkUpgradeConfigMessage (as serialized from
-// UpgradeConfigToNetworkMessage).
+// Attempts to parse a params.UpgradeConfig from a []byte
 //
 // The function returns a reference of *params.UpgradeConfig
-func NewUpgradeConfigMessageFromBytes(bytes []byte) (*params.UpgradeConfig, error) {
-	var config networkUpgradeConfigMessage
-	version, err := Codec.Unmarshal(bytes, &config)
+func UpgradeConfigFromBytes(bytes []byte) (*params.UpgradeConfig, error) {
+	var upgradeConfig params.UpgradeConfig
+	err := json.Unmarshal(bytes, &upgradeConfig)
```

Comment:
We should comment that this only works for JSON upgrade bytes. There is no requirement that JSON is used here generally.

Reply:
I agree @patrick-ogrady, this was just a quick prototype. The alternatives are described in #1020 or relying in an external library such as [json-canonicalization](https://github.com/cyberphone/json-canonicalization).

For me the ideal solution would be to extend our codec to allow the serialization of structs not owned by the package.

Reply:
There are many other options to calculate a hash, but I've been told to avoid reflection at all costs.